classes/profile.php

```diff
@@ -16,35 +16,41 @@
 
 namespace tool_excimer;
 
-defined('MOODLE_INTERNAL') || die();
-
 /**
- * Functions relevant to profiles.
+ * Profile saving/loading and manipulation.
  *
  * @package   tool_excimer
- * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>, Kevin Pham <kevinpham@catalyst-au.net>
  * @copyright 2021, Catalyst IT
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 class profile {
+    const TABLE = 'tool_excimer_profiles';
+
 
-    /** Request's fallback value for when the $SCRIPT is null */
-    const REQUEST_UNKNOWN = 'UNKNOWN';
+    /** Reason - MANUAL - Profiles are manually stored for the request using FLAMEME as a page param. */
+    const REASON_MANUAL   = 0b0001;
 
-    /** Report section - recent - lists the most recent profiles first */
-    const REPORT_SECTION_RECENT = 'recent';
+    /** Reason - SLOW - Set when conditions are met and these profiles are automatically stored. */
+    const REASON_SLOW     = 0b0010;
 
-    /** Report section - slowest - lists the slowest profiles first */
-    const REPORT_SECTION_SLOWEST = 'slowest';
+    /** Reason - FLAMEALL - Toggles profiling for all subsequent pages, until FLAMEALLSTOP param is passed as a page param. */
+    const REASON_FLAMEALL = 0b0100;
 
-    /** Report section - unfinished - lists profiles of scripts that did not finish */
-    const REPORT_SECTION_UNFINISHED = 'unfinished';
+    /** Reason - NONE - Default fallback reason value, this will not be stored. */
+    const REASON_NONE = 0b0000;
```

Comment:
Can you put these in order so 0 would be first

Reply:
done